File: scenarios/long_chain_dag/fixtures/read_markers.py
from __future__ import annotations

import json
import sys
from pathlib import Path


def main() -> None:
    workspace = Path(sys.argv[1])
    markers = sorted(path.stem for path in workspace.glob("*.txt"))
    print(json.dumps({"lineage": markers, "summary": {"status": "success"}}))


if __name__ == "__main__":
    main()

File: scenarios/long_chain_dag/fixtures/read_markers.ts
import { readdir } from "node:fs/promises";
import path from "node:path";

const workspace = process.argv[2];
if (!workspace) {
  throw new Error("workspace is required");
}

const markers = (await readdir(workspace))
  .filter((name) => name.endsWith(".txt"))
  .map((name) => path.basename(name, ".txt"))
  .sort();

console.log(JSON.stringify({ lineage: markers, summary: { status: "success" } }));
